sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -67,7 +67,8 @@ trait LegacyDateFormatter extends DateFormatter {
 
   override def parse(s: String): Int = {
     val micros = DateTimeUtils.millisToMicros(parseToDate(s).getTime)
-    DateTimeUtils.microsToDays(micros)
+    val julianDays = DateTimeUtils.microsToDays(micros)
```

Comment:
I think need to rebase micros instead of days because `microsToDays()` is based on Proleptic Gregorian calendar already.

Reply:
Thanks @MaxGekk, I made the change.